documents/DeveloperGuide/MainPage.md

```diff
@@ -2,25 +2,213 @@
+
+You can build MaterialX using any of the following methods:
+
+1. [CMake GUI](#cmake-gui)
+2. CMake Command-Line Interface (CLI)
+   - [Passing options as arguments](#pass-options-as-arguments)
+   - [Specifying options in a CMake preset](#specify-options-in-a-cmake-preset)
+3. [Using an IDE](#use-an-ide)
+
+There is no recommended method; it’s purely based on personal preference.
+
+### CMake GUI
+You can use the CMake GUI to configure and generate project files for MaterialX. Note that CMake GUI only generates build files—the actual build must be performed in an external tool such as an IDE or the terminal.
+
+![CMake GUI Example](../Images/cmake_gui_example.png)
+
+To get started, open CMake GUI and go through these steps:
+
+1. **Browse Source**: Select the root of the cloned MaterialX repository.
+2. **Browse Build**: Choose a build directory (e.g. `MaterialX/build`).
+3. Click **Configure**: Click Configure: This saves your current configuration. You'll need to click it again anytime you change any options. The first time you click it, CMake will prompt you to choose a generator (e.g., Visual Studio, Ninja).
+4. Click **Generate**: This step creates the build system files (e.g., a Visual Studio solution or Makefiles).
+
+You’ll see progress messages in the output window confirming each step. For example:
+```
+Configuring done (3.9s)
+Generating done (1.0s)
+```
+
+After generation, you have two options to build the project:
+- Click **Open Project** to launch the generated project in your default IDE or some other configured environment to open your project, then build from there.
+- Or, build directly from the terminal:
+```bash
+cd <your-build-directory>
+cmake --build .
+```
+
+> [!Note]
+> The CMake GUI includes **Grouped** and **Advanced** checkboxes, which control how the environment variables are displayed in the UI. Use them to manage build configuration more easily.
+
+### CMake CLI
+
+The CMake Command-Line Interface (CLI) offers several ways to configure and build the project, as outlined below.
+
+#### Pass Options as Arguments
 
-## Supported Platforms
+You can specify build options directly during the CMake configuration step using the `-D` flag. Reference the [YAML build actions](../../.github/workflows/main.yml) in the repository for examples.
 
-The MaterialX codebase requires a compiler with support for C++17, and can be built with any of the following:
+Example:
 
-- Microsoft Visual Studio 2017 or newer
-- GCC 8 or newer
-- Clang 5 or newer
+```bash
+cd MaterialX
+cmake -S . -B build -DMATERIALX_BUILD_VIEWER=ON -DMATERIALX_BUILD_GRAPH_EDITOR=ON
+cmake --build ./build
+```
+
+> [!Note]
+> Command syntax may vary depending on your tools. For example, with **Ninja + Clang**:
+>
+> ```bash
+> cmake -G "Ninja" -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -S . -B build -DMATERIALX_BUILD_VIEWER=ON -DMATERIALX_BUILD_GRAPH_EDITOR=ON
+> cmake --build build
+> ```
+
+
+#### Specify Options in a CMake Preset
```

Comment:
This section titled `Specify Options in a CMake Preset` seems slightly too niche for our Developer Guide, and delays the introduction of the key section on `Building MaterialX` below.  Perhaps we should omit it?